wolfcrypt/src/poly1305.c

```diff
@@ -1222,7 +1222,7 @@ int wc_Poly1305SetKey(Poly1305* ctx, const byte* key, word32 keySz)
     printf("\n");
 #endif
 
-    if (keySz != 32 || ctx == NULL)
+    if (keySz != 32 || ctx == NULL || key == NULL)
```

Comment:
The check for "key == NULL" should be above the previous code block for CHACHA_AEAD_TEST, as that code uses "key" and would seg fault / crash if a NULL key pointer was used.